lib/php/lib/Transport/TSocket.php

```diff
@@ -253,7 +253,9 @@ public function open()
         if (function_exists('socket_import_stream') && function_exists('socket_set_option')) {
             // warnings silenced due to bug https://bugs.php.net/bug.php?id=70939
             $socket = @socket_import_stream($this->handle_);
-            @socket_set_option($socket, SOL_TCP, TCP_NODELAY, 1);
+            if ($socket !== false) {
```

Comment:
previous function can return false, and in php 8.0 it trigger a type error, so i added a check.

In future it will be great to remove all @ and work correctly with all warning and error generated by stream functions